OneCommunityGlobal/HighestGoodNetworkApp
# Chris update equipment frontend

# Description
Implemented the update equipment page in the bmdashboard as part of the phase 2 development.
<img width="771" alt="Screenshot 2024-04-14 at 8 23 28 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114450390/f45f34be-102f-4289-8c46-3b9e33a9323c">


## Related PRS (if any):
This frontend PR is not related to any backend pr, just go into development branch of HNG
…

## Main changes explained:
Created UpdateEquipment.jsx and UpdateEquipment.css
Added routing for update equipment
Added the url and added pages like equipmentactions, equipmentreducers, equipmentconstants etc
…

## How to test:
1. check into current branch
2. do npm install and ... to run this PR locally(npm start)
3. Clear site data/cache
4. log as any user eg: volunteer, owner or admin
5. go to http://localhost:3000/bmdashboard and login again
6. Navigate to http://localhost:3000/bmdashboard/tools/65c16ef325834114ac9427e0/update
7. Verify the design from the figma template https://www.figma.com/file/kpwXZP2pLtOJ1Fn2u7Sivn/HGN-PII?type=design&node-id=404-6751&mode=design&t=WKyVjVeztidxL1Y9-0
8. Verify the details of the page from the database from buildingInventoryItems collection.
9. Feel free to change the id in the url to any other id in the database (the type has to be equipment) and verify the webpage data.


## Screenshots or videos of changes:
![Screenshot 2024-04-14 at 8 20 15 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114450390/d3cd11e4-88a5-4da1-b036-6914446446c3)
![Screenshot 2024-04-14 at 8 20 54 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114450390/2bf4ee91-ec3f-4c66-9f5e-1e8d05163a4e)


## Note:
The scope of this pr is just limited to the design of retrieving the equipment data related to its picture, name, number, class, project, current status, ownership, rental end date, and days left. All the other data that need to be updated are not working because there isn't fields in the database to update those data to, once we are done with creating those fields, we can actually update the information into the database. So for now, WHEN YOU HIT SUBMIT, NOTHING WILL HAPPEN.
